### PR TITLE
clean up the cue bar component

### PR DESCRIPTION
**Pull request by joelhooks**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "browser-cookies": "^1.2.0",
     "client-oauth2": "^4.3.3",
     "cookie": "^0.4.1",
-    "cueplayer-react": "^0.3.0",
+    "cueplayer-react": "^0.3.1",
```
Comment (joelhooks) [inline package.json:64]: move the text track loading to the player lib

File: src/components/player/cue-bar.tsx [hunk 2]
```diff
@@ -1,82 +1,54 @@
...
-import {isEqual} from 'lodash'
 
-export default class CueBar extends React.Component<any> {
-  state: any = {cues: []}
-  constructor(props: any) {
-    super(props)
-    this.state = {cues: this.getTextTrackItems(), currentCuePopup: null}
-  }
+const CueBar: React.FC<any> = ({className, disableCompletely, player}) => {
+  const {duration, activeMetadataTracks} = player
```
Comment (joelhooks) [inline src/components/player/cue-bar.tsx:5]: now the "notes" tracks live in the player state

File: src/components/player/cue-bar.tsx [hunk 3]
```diff
@@ -1,82 +1,54 @@
...
 import classNames from 'classnames'
-import {isEqual} from 'lodash'
 
-export default class CueBar extends React.Component<any> {
-  state: any = {cues: []}
-  constructor(props: any) {
-    super(props)
-    this.state = {cues: this.getTextTrackItems(), currentCuePopup: null}
-  }
+const CueBar: React.FC<any> = ({className, disableCompletely, player}) => {
```
Comment (joelhooks) [inline src/components/player/cue-bar.tsx:4]: no need for this to be a class

File: src/components/player/cue-bar.tsx [hunk 4]
```diff
@@ -1,82 +1,54 @@
...
-  getTextTrackItems() {
-    const {kinds = ['metadata'], player} = this.props as any
-    const {textTracks} = player
-    const textTrackItems: any = {}
-    const tracks = Array.from(textTracks || [])
+export default CueBar
 
-    if (tracks.length === 0) {
-      return textTrackItems
+const useCue = (cue: VTTCue) => {
```
Comment (joelhooks) [inline src/components/player/cue-bar.tsx:26]: small hook for doing the even listening on a single cue

File: src/components/player/cue-bar.tsx [hunk 5]
```diff
@@ -1,82 +1,54 @@
...
-        })}
-      </div>
-    )
-  }
+  return [active, setActive]
 }
 
-const CuePopup: React.FC<any> = ({note, popupPersists, currentCuePopup}) => {
-  return note?.title === currentCuePopup || popupPersists ? (
+const CuePopup: React.FC<any> = ({cue, active}) => {
```
Comment (joelhooks) [inline src/components/player/cue-bar.tsx:49]: is it active? that's all that it really needs to know for display

File: src/components/player/cue-bar.tsx [hunk 6]
```diff
@@ -147,23 +91,20 @@ const NoteCue: React.FC<any> = ({
       )}
       style={{left: startPosition}}
       onClick={() => {
-        setCurrentCuePopup(
-          note?.title === currentCuePopup && popupPersists ? null : note?.title,
-        )
-        setPopupPersists(!popupPersists)
+        if (active && !persist) {
```
Comment (joelhooks) [inline src/components/player/cue-bar.tsx:94]: consolidated logic for display here for now

**Review by DrShpongle — APPROVED** [review]
Looks cleaner!

**Pull request merged**